packages/extension-commerce-bm-seo/src/setup-app.ts

```diff
@@ -92,22 +92,32 @@ class CommerceBmSeo extends ApplicationExtension<Config> {
         // Make SEO GET Url Mapping API call
         const config = this.getConfig()
         const shopperSeo = await getShopperSeoClient(locals, config)
-        const urlMapping = await shopperSeo.getUrlMapping({
+        let urlMapping = await shopperSeo.getUrlMapping({
             parameters: {urlSegment: locals.originalUrl}
         })
         console.log('--- url mapping', urlMapping)
 
+        // DEBUG
+        urlMapping = {resourceType: 'category'}
+
         if (!urlMapping.resourceType) {
             return Promise.resolve([])
         }
 
         // Transform URL mapping to serialized route
         const requestURL = new URL(locals.originalUrl, getAppOrigin(locals))
         const componentName =
-            config.resourceTypeToComponentMap[urlMapping.resourceType as keyof typeof config.resourceTypeToComponentMap]
+            config.resourceTypeToComponentMap[
+                urlMapping.resourceType as keyof typeof config.resourceTypeToComponentMap
+            ]
+
         return Promise.resolve([
             {
                 path: requestURL.pathname,
+                // Placeholder component that will be replaced later during beforeRouteMatch
+                component: Sample,
```

Comment:
do we even need the placeholder component in this approach?

I wonder if we should just let extension developers implement `getRoutesAsync` in 2 ways:
1. returns a `RouteProps` with a `component` that is defined in the extension
2. returns a `RouteProps` with a `componentName` that is defined in another extension